hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java

```diff
@@ -268,6 +275,12 @@ public Pipeline createPipeline(ReplicationType type,
     }
     lock.writeLock().lock();
     try {
+      if (freezePipelineCreation.get()) {
+        LOG.debug("Pipeline creation is frozen while an upgrade is in " +
```

Comment:
Do we want to throw an exception here? The background pipeline creator will keep trying to create pipelines, and the logs may be filled with exceptions. Is there a way to do this more cleaner? Maybe something like

_Stop pipeline creator --> Add flag in pipeline manager to stop createPipeline (for manual pipeline creation) -> Destroy existing pipelines_

Reply:
OZONE_SCM_PIPELINE_CREATION_INTERVAL_DEFAULT is 2 minutes. I thought that upgrade finalization should be few minutes. Its possible that we may have few messages in the log to indicate that pipeline creation is frozen due to upgrade. But I can add a "pause" for pipeline creator as suggested.